addons/account_asset/models/account_asset.py

```diff
@@ -102,38 +102,27 @@ def _compute_entry_count(self):
 
     @api.depends('value', 'salvage_value', 'depreciation_line_ids.move_check', 'depreciation_line_ids.amount')
     def _compute_amount_residual(self):
-        total_amount = 0.0
-        for line in self.depreciation_line_ids:
-            if line.move_check:
-                total_amount += line.amount
-        self.value_residual = self.value - total_amount - self.salvage_value
+        for asset in self:
```

Comment:
If you do some refactoring ... total_amount = sum(asset.depreciation_line_ids.filtered(lambda line: line.move_check).mapped('amount') ?